### PR TITLE
7768 - Fix unit tests for multimedia forms

### PR DESCRIPTION
**Pull request by latin-panda**
# Description

This PR fixes unit tests for multimedia forms.
- Adds validation to prevent navigation when using `windows.history` and `form.pages._getCurrentIndex()` returns undefined. This mainly because we want to test the feature and karma doesn't do well on navigation. Also, we can't mock `windows.history` because it's ready only. 
- Enables unit tests that were breaking karma

https://github.com/medic/cht-core/issues/7768

Screen Shot 2022-09-12 at 3 12 19 pm

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by latin-panda** [comment]
@dianabarsan Can you please have a look? I will update the branch once the e2e fix is in master. 
Thanks!

File: webapp/src/ts/services/enketo.service.ts [hunk 1]
```diff
@@ -340,8 +340,10 @@ export class EnketoService {
           .then(valid => {
             if (valid) {
               const currentIndex = form.pages._getCurrentIndex();
-              window.history.pushState({ enketo_page_number: currentIndex }, '');
-              this.setupNavButtons($wrapper, currentIndex);
+              if (currentIndex !== undefined) {
```
Comment (dianabarsan) [inline webapp/src/ts/services/enketo.service.ts:343]: I'm usually reluctant to change app code just to overcome test limitations. Is there a way mock `_getCurrentIndex` so this change isn;t necessary?

Comment (latin-panda) [reply webapp/src/ts/services/enketo.service.ts:343]: even if I make it return something, the `window.history` executes and karma doesn't like it

Comment (latin-panda) [reply webapp/src/ts/services/enketo.service.ts:343]: actually it's just the `window.history.back`, 1 sec. I'm going to push a second version

Comment (latin-panda) [reply webapp/src/ts/services/enketo.service.ts]: Okay, I've reverted the validation and now it just updates the test file. Pushed!
Screen Shot 2022-09-12 at 6 30 22 pm

**Review by dianabarsan — APPROVED** [review]
Awesome! Thanks for the fix!

**Pull request merged**